File: python/moveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        nonZeroPointer = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[nonZeroPointer] = nums[i]
                nonZeroPointer += 1
        for i in range(nonZeroPointer, len(nums)):
            nums[i] = 0

File: i_algorithms/move_zeroes.cpp
#include <vector>
using namespace std;
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int nz = 0;
        for(int i=0;i<nums.size();i++){
            if(nums[i]!=0){
                nums[nz]=nums[i];
                nz++;
            }
        }
        for(int i=nz;i<nums.size();i++){
            nums[i]=0;
        }
    }
};